Subject: [PATCH] ChooseLockGeneric: fix crash

* getActivity() returns null if activity has been finished already

Steps to reproduce:

 1. Set a pattern
 2. Add a fingerprint
 3. Set screen lock to none/swipe
 4. Set a pattern again
 5. Add a fingerprint

diff --git a/src/com/android/settings/ChooseLockGeneric.java b/src/com/android/settings/ChooseLockGeneric.java
index 3b592d4..95fd0e6 100644
--- a/src/com/android/settings/ChooseLockGeneric.java
+++ b/src/com/android/settings/ChooseLockGeneric.java
@@ -787,8 +787,13 @@
         }
 
         private void removeManagedProfileFingerprintsAndFinishIfNecessary(final int parentUserId) {
+            final Activity activity = getActivity();
+            if (activity == null) { // Activity has been detached, so just finish it
+                finish();
+                return;
+            }
             mFingerprintManager.setActiveUser(UserHandle.myUserId());
-            final UserManager um = UserManager.get(getActivity());
+            final UserManager um = UserManager.get(activity);
             boolean hasChildProfile = false;
             if (!um.getUserInfo(parentUserId).isManagedProfile()) {
                 // Current user is primary profile, remove work profile fingerprints if necessary
